### ISSUE TITLE
renaming "test" split to "dev"

### ISSUE THREAD

**Issue by felixbur** [felixbur/nkululeko#59]
Nkululeko only knows two splits: train and test.
but it would be more correct to name the "test" split "dev" (short for development), as  we kind of always use it to optimize a model.
Any thoughts?

**Comment by bagustris** [comment]
I think It is already implemented, but I checked again, and It is not yet.

The split of datasets under `data` already defines their dev sets (e.g., ravdess, crema-d). So. that split ("dev") was not used during model training? If so, maybe this is the cause of why the results (e.g., UAR) were below what I expected.

CMIIW, "dev" set actually is needed only for neural-network-based models (currently MLP, MLP-REG, CNN). For others like SVM, KNN, we can concatenate "train" and "dev" into "train". 

So, IMHO, there will be three sets: train, dev, and test. For models other than mlp and cnn, train and dev will concatenated for train set. For unseen data (hidden test), we can make inferences/predictions from the model (already implemented partly?) to predict the target.

**Comment by bagustris** [comment]
@felixbur 
I think it makes sense. See my two cases above.
In case 1, the current 'test' split should be renamed to 'dev' split.
In case 2, we need to reintroduce the 'test' split again. 

So, consider the following (it is common, e.g., in ComParE challenge). There are three splits given by the authors of the dataset: `train.csv`, `dev.csv`, and `test.csv`. Train and dev have labels, the test does not (in the CSV file, it only contains a file, the labels (`e.g.` emotion), are usually replaced by a question mark. 

So there are possibilities for building a model:  
- using `train` only, evaluation metric for training
- using `train` + `dev`, evaluation metric for dev
- using `train` +`dev` for training and `test`, evaluation metric for training

The output in the last option will be a CSV file containing a file and prediction of labels. This file usually is submitted to the organizer to obtain the score of the test set.

**Comment by bagustris** [comment]
A simple workaround maybe just keep the current test as it is but providing more option if test has label (default) to differ where is it dev (has label) or test (unseen).

```INI
[DATA]
test.has_labels = False
```

By default, it assumed test has labels (`test.has_labels = True`), if it hasn't

**Comment by bagustris** [comment]
@felixbur 
After realizing that Nkululeko already has `.test` and `.demo` modules, this proposal absolutely makes sense (renaming test to dev).

One suggestion point that after getting the best model, user will be allowed to use both `train` and `dev` data to `train` so the final model contains more training data.

**Comment by felixbur** [comment]
closed for https://github.com/felixbur/nkululeko/issues/184
